lib/ansible/modules/packaging/os/satellite_subscription.py

```diff
@@ -0,0 +1,441 @@
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {"metadata_version": "1.1", "status": ["preview"], "supported_by": "community"}
+
+
+DOCUMENTATION = """
+---
+module: satellite_subscription
+short_description: Red Hat Satellite subcription (license) mangement module
+description:
+  - This module allows to assign or remove subscriptions to hosts in the satellite
+  - It works on the Satellite side, so not from the client side like redhat_subscription.
+    Your hosts or hypervisors should be in the Satellite already. The module will run subscription
+    related tasks (assign, remove, autoattach or replace with hypervisor subscriptions) for the
+    hosts through the Satellite API.
+version_added: "2.8"
+requirements:
+  - requests
+author:
+  - Luc Stroobant (@stroobl)
+options:
+  satellite_hostname:
+    description:
+      - Red Hat Satelite server (6.x) hostname
+    required: true
+  username:
+    description:
+      - Satellite server username
+    required: true
+  password:
+    description:
+      - Satellite server password
+    required: true
+  organization:
+    description:
+      - Satellite organization ID to work on
+    required: true
+  state:
+    description:
+      - "present: assign a subscription of subscription_type to host"
+      - "absent: remove subscription of subscription_type from host"
+      - "autoattach: run subscription autoattach for host (autoattach will be enabled if necessary)"
+      - "disable_autoattach: disable subscription autoattach for host"
+      - "vdcguests: remove all Red Hat subscriptions from a hypervisor's guests and run autoattach
+         to replace them with virtual guest subscriptions. Run this against a virt-who hypervisor to
+         remove all other subscriptions assigned to the guests. Useful if you configured a VDC
+         subscription on a hypervisor that didn't have it before."
+      - "optimize: use to replace subscriptions on physical servers with a higher multiplier with subscriptions with a lower multiplier."
+    choices: [present, absent, autoattach, disable_autoattach, vdcguests, optimize]
+    default: autoattach
+  content_host:
+    description:
+      - List of content hosts to work with
+      - Can also be a hostname search pattern accepted by the satellite.
+  subscription:
+    description:
+      - List of subscriptions to assign or remove
+      - Required when action is assign or remove
+      - "This field is used for the search for subscriptions, it can contain (part of) the subscription
+         name, eg 'Red Hat Enterprise Linux Server Entry Level, Self-support' or another search query
+         as used in the satellite, eg: 'product_id=RH00005S'"
+  unique:
+    description:
+     - Set to true to remove all other subscriptions attached to the host
+    type: bool
+    default: False
```

Comment:
The documentation uses yes/no as boolean values. So the documentation should reflect that.
```suggestion
    default: no
```